### PR TITLE
Add del2,del4 source terms to manufactured solution

### PR DESCRIPTION
**Pull request by cbegeman**
When either or both of del2 and del4 momentum terms are included in the dynamics, add the corresponding source terms to the manufactured solution.

### CODE REVIEW

**Comment by cbegeman** [comment]
@sbrus89 Can you take a look at the del4 source terms and let me know what you think? I'm getting noisy errors and wondering if div_factor = 1 isn't equivalent to \nabla^4 **u**. The del2 term is checking out fine.

File: components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F [hunk 1]
```diff
@@ -183,6 +184,19 @@ subroutine ocn_manufactured_solution_tend_vel(tend, err)!{{{
                       + ang_freq*sin(phase) &
                       - 0.5_RKIND*eta0*(kx + ky)*sin(2.0_RKIND*(phase))) 
 
+            if (.not. config_disable_vel_hmix) then
+               if (config_use_mom_del2) then 
+                  viscDel2 = config_mom_del2
+                  u = u + viscDel2 * eta0 * kx**2 * cos(phase)
+                  v = v + viscDel2 * eta0 * ky**2 * cos(phase)
```
Comment (hyungyukang) [inline components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F:191]: I would like to suggest to modify Del2 source terms:

```suggestion
                  u = u + viscDel2 * eta0 * (kx**2 + ky**2) * cos(phase)
                  v = v + viscDel2 * eta0 * (kx**2 + ky**2) * cos(phase)
```

File: components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F [hunk 2]
```diff
@@ -183,6 +184,19 @@ subroutine ocn_manufactured_solution_tend_vel(tend, err)!{{{
...
+               endif
+               if (config_use_mom_del4) then 
+                  viscDel4 = config_mom_del4
+                  u = u - viscDel4 * eta0 * (kx**4 * cos(phase) + kx**2 * ky**2 * cos(phase))
+                  v = v - viscDel4 * eta0 * (ky**4 * cos(phase) + kx**2 * ky**2 * cos(phase))
```
Comment (hyungyukang) [inline components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F:196]: As well as Del4 source terms:
```suggestion
                  u = u - viscDel4 * eta0 * ((kx**4 + ky**4 + kx**2 * ky**2) * cos(phase))
                  v = v - viscDel4 * eta0 * ((kx**4 + ky**4 + kx**2 * ky**2) * cos(phase))
```

File: components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F [hunk 3]
```diff
@@ -58,6 +58,7 @@ module ocn_manufactured_solution
    real (kind=RKIND) :: kx, ky
    real (kind=RKIND) :: ang_freq
    real (kind=RKIND) :: eta0
+   real (kind=RKIND) :: viscDel2, viscDel4
```
Comment (sbrus89) [inline components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F:61]: These changes look great. One very minor comment is that these could be set in the init routine.

Comment (cbegeman) [reply components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F:61]: Good point. Done

**Comment by cbegeman** [comment]
@hyungyukang and @sbrus89 I have made the requested changes and tested del2 and del4 with the polaris test case. Can you re-review whenever it's convenient? Thanks!

**Review by sbrus89 — APPROVED** [review]
@cbegeman, this looks great. I think it's ready to move over to an E3SM PR.

**Review by hyungyukang — APPROVED** [review]
@cbegeman , the changes look great. Thanks a lot!

**Comment by cbegeman** [comment]
@hyungyukang and @sbrus89 thank you both so much for your reviews, especially @hyungyukang for finding those bugs!

**Comment by cbegeman** [comment]
Closed in favor of https://github.com/E3SM-Project/E3SM/pull/6862

**Pull request closed without merging**